neo4j/result_helpers.go

```diff
@@ -78,19 +79,32 @@ func Single(result Result, err error) (*Record, error) {
 // Collect loops through the result stream, collects records into a slice and returns the
 // resulting slice. Any error passed in or reported while navigating the result stream is
 // returned without any conversion.
-//   records, err := neo4j.Collect(session.Run(...))
+//
+//	records, err := neo4j.Collect(session.Run(...))
+//
+// Deprecated: use CollectWithContext instead (the entry point of context-aware
+// APIs is NewDriverWithContext)
 func Collect(result Result, err error) ([]*Record, error) {
 	if err != nil {
 		return nil, err
 	}
 	return result.Collect()
 }
 
+// CollectWithContext behaves the same as Collect
```

Comment:
Maybe move the main documentation here as you want users who use the new method to have an easy time.
Instead refer to the docs of `CollectWithContext` in the docs of `Collect`.

Reply:
Fixed in 56dc354854459b43e676cbc21baaf457ac54d477